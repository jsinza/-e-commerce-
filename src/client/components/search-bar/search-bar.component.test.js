import React from 'react';
import { shallow } from 'enzyme';
import 'babel-polyfill';
import * as router from 'react-router-dom';
import SearchBar from './search-bar.component';

const PushMock = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));

describe('SearchBar Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchBar />);
    router.useHistory.mockReturnValue({ push: PushMock });
  });
  it('should render SearchBar Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call redirect when enter is pressed', () => {
    wrapper.find("[className='search-bar__search']").props().onKeyUp({ keyCode: 13 });
    expect(PushMock).toHaveBeenCalledTimes(1);
  });

  it('should redirect to the url of items with the search query', () => {
    wrapper.find("[className='search-bar__search']").props().onChange({
      target: {
        value: 'zapatos',
      },
    });
    wrapper.find("[className='search-bar__search']").props().onKeyUp({ keyCode: 13 });
    expect(PushMock.mock.calls[1][0]).toBe('/items?q=zapatos');
  });

});
