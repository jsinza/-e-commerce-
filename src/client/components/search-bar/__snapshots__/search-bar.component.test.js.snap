// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should render SearchBar Component 1`] = `ShallowWrapper {}`;
